src/Controller/DownloadController.php

```diff
@@ -108,26 +108,22 @@ public function defaultAction(int $id)
     public function httpAction(Dataset $dataset, Datastore $dataStore, LogActionItemEventDispatcher $logActionItemEventDispatcher)
     {
         $datasetSubmission = $dataset->getDatasetSubmission();
-        $response = new Response(
-            null,
-            Response::HTTP_NO_CONTENT
-        );
         if ($datasetSubmission instanceof DatasetSubmission) {
             $fileset = $datasetSubmission->getFileset();
 
             if ($fileset instanceof Fileset) {
                 if (!$fileset->doesZipFileExist()) {
                     $filePhysicalPath = $fileset->getProcessedFiles()->first()->getPhysicalFilePath();
-                    $response = new StreamedResponse(function () use ($dataStore, $filePhysicalPath) {
-                        $outputStream = fopen('php://output', 'wb');
+                    try {
                         $fileStream = $dataStore->getFile($filePhysicalPath);
-                        if (is_array($fileStream)) {
-                            stream_copy_to_stream($fileStream['fileStream'], $outputStream);
-                        } else {
-                            throw new BadRequestHttpException('Unable to open file');
-                        }
+                    } catch (\Exception $exception) {
+                        throw new BadRequestHttpException('Unable to open file');
+                    }
+                    $response = new StreamedResponse();
+                    $response->setCallback(function () use ($fileStream) {
```

Comment:
You could probably put the callback back in the constructor, but this works too.

Reply:
Either way I'm ok